pygmt/tests/test_clib_read_data.py

```diff
@@ -0,0 +1,223 @@
+"""
+Test the Session.read_data method.
+"""
+
+import ctypes as ctp
+from pathlib import Path
+
+import numpy as np
+from pygmt.clib import Session
+from pygmt.datatypes import _GMT_DATASET, _GMT_GRID, _GMT_IMAGE
+from pygmt.helpers import GMTTempFile
+
+
+def test_clib_read_data_dataset():
```

Comment:
This test is a simplified version of the doctest in `pygmt/datatypes/dataset.py`.